signer/src/testing/dummy.rs

```diff
@@ -510,6 +512,48 @@ impl fake::Dummy<fake::Faker> for SigHash {
     }
 }
 
+impl fake::Dummy<std::ops::Range<u64>> for BitcoinBlockHeight {
+    fn dummy_with_rng<R: rand::Rng + ?Sized>(config: &std::ops::Range<u64>, rng: &mut R) -> Self {
+        // Overwriting config because values greater then i64::MAX will cause db conversion errors.
+        let mut config = config.clone();
+        if config.end > i64::MAX as u64 || config.start >= config.end {
+            config.start = 0;
+            config.end = i64::MAX as u64;
+        }
+        rng.gen_range(config).into()
```

Comment:
If `end > i64::MAX`, why should the start then be fixed to zero? Why not just do:
```rust
rng.gen_range(config).min(i64::MAX as u64).into()
```
Well, that panics if the range is empty but that is probably the correct thing to do. Same for `StacksBlockHeight`

Reply:
My logic was following:

- Let's fallback to uniform distribution 0..i64max in case we have invalid config
- Why don't return err/panic on bad config? Mostly because my impl of Dummy<Faker> will generate invalid config in 50% cases
- Why I'm ok with all of that??? 1) In most cases I want height to be just random number (and fallback to unform on 0..i64::max is perfect here). 2) If in some cases I want some non-default random then this range based dummy should work well for me. I don't see anything else I want from Dummy impl of heights

Reply:
I see. Well, I don't think that the `Dummy<Faker>` impl should fail. And even for this one, we probably shouldn't take the `min` as I was suggesting.

Reply:
Since this is a version using an explicit range, if the provided range is wrong it wouldn't be too unexpected to panic for invalid ones. But we could silently cap the range for ergonomics (given we use `Range<u64>`, so maybe we end up with some automatic `u64::MAX` that would be annoying to explicitly type).
But regardless of that choice, I also find it strange to fallback to `0..i64max` for *any* invalid config.

Reply:
Yeah `min(i64::max)` would be bad because you'll end up with a whole bunch of `i64::MAX`.

Why not just use the u64 range directly (for this explicit impl)? There's no invariant on the block height itself, it's postgres that forces i64 on us. This is how it worked before in the dummy attributes, and yes i've caused it to panic by forgetting to limit it to i64::max... but that was my fault 😊

I'm not sure there's any usages left after the attribute removals in models.rs anyway, so we may be able to just drop these impls too?

Reply:
> I'm not sure there's any usages left after the attribute removals in models.rs anyway, so we may be able to just drop these impls too?

Good point, I think that we should drop these impls if we aren't using them right now. If we need them later for something specific then we can add them back.

Reply:
Yeah, agree, I like decision to remove range-config impl and just do range(0..i64max) on dummy config impl. Implemented here: [8b76743](https://github.com/stacks-sbtc/sbtc/pull/1448/commits/8b76743e5e1b1b3a0dd12c2cc8d327cd815fe74e)